src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/utils/CancellationProcessValidator.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.controllers.utils;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.wataskmanagementapi.auth.access.entities.AccessControlResponse;
+import uk.gov.hmcts.reform.wataskmanagementapi.config.LaunchDarklyFeatureFlagProvider;
+import uk.gov.hmcts.reform.wataskmanagementapi.config.features.FeatureFlag;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+@Component
+public class CancellationProcessValidator {
+
+    private LaunchDarklyFeatureFlagProvider launchDarklyFeatureFlagProvider;
+
+    @Autowired
+    public CancellationProcessValidator(LaunchDarklyFeatureFlagProvider launchDarklyFeatureFlagProvider) {
```

Comment:
For future reference, we can use lombok.AllArgsConstructor instead of this constructor (unless you need it for specific testing purposes)